autoreporter_addons/ssl_checks/reporter.py

```diff
@@ -0,0 +1,203 @@
+    CERTIFICATE_AUTHORITY_INVALID = ReportType("certificate_authority_invalid")
+    NO_HTTPS_REDIRECT = ReportType("no_https_redirect")
+    BAD_CERTIFICATE_NAMES = ReportType("bad_certificate_names")
+    ALMOST_EXPIRED_SSL_CERTIFICATE = ReportType("almost_expired_ssl_certificate")
+    EXPIRED_SSL_CERTIFICATE = ReportType("expired_ssl_certificate")
+
+    @staticmethod
+    def create_reports(task_result: Dict[str, Any], language: Language) -> List[Report]:
+        if task_result["headers"]["receiver"] != "ssl_checks":
+            return []
+
+        domain = task_result["payload"]["domain"]
+        domain_parts = [part for part in domain.split(".") if part]
+        # We do the filtering both in the scanning module and in the reporter so that after changing this setting
+        # the user wouldn't need to wait for the next scan for the configuration to take effect.
+        if domain_parts[0] in ExtraModulesConfig.SUBDOMAINS_TO_SKIP_SSL_CHECKS:
+            return []
+
+        if not isinstance(task_result["result"], dict):
+            return []
+
+        try:
+            response = cached_get(f"https://{domain}")
+            parent_response = cached_get(f"https://{'.'.join(domain_parts[1:])}")
+            if SequenceMatcher(None, response.content, parent_response.content).quick_ratio() >= 0.8:
+                # Do not report misconfigurations if a domain has identical content to a parent domain - e.g.
+                # if we have mail.domain.com with identical content to domain.com, we assume that it's domain.com
+                # which is actually used, and therefore don't report subdomains.
+                return []
+        except Exception:
+            logger.warning(f"Unable to check whether domain {domain} has identical content to parent domain")
+
+        if "response_status_code" in task_result["result"] and "response_content_prefix" in task_result["result"]:
+            response_status_code = task_result["result"]["response_status_code"]
+            response_content_prefix = task_result["result"]["response_content_prefix"]
+
+            if (
+                response_status_code == 404
+                or response_status_code == 403
+                # This one is important - sometimes we reported false positives after getting a 5xx error (and thus no redirect)
+                or (response_status_code >= 500 and response_status_code <= 599)
+                or "<html" not in response_content_prefix.lower()
+                or any([fragment in response_content_prefix for fragment in FILTERED_WEBSITE_FRAGMENTS])
+            ):
+                # Not something actually usable, won't be reported
+                return []
+
+            # We have systems in our constituency where a redirect is performed from http://domain to a different domain and https://domain is not used
+            # and is misconfigured/the certificate expires. For now let's treat this as a false positive.
```

Comment:
get `task_result["result"]` and `task_result["payload"]` as local vars, should clean code a little

Reply:
done